ttnn/CMakeLists.txt

```diff
@@ -449,3 +449,74 @@ set_target_properties(ttnn PROPERTIES
     CXX_VISIBILITY_PRESET "default"
     ADDITIONAL_CLEAN_FILES "${PROJECT_SOURCE_DIR}/ttnn/ttnn/_ttnn.so;${PROJECT_SOURCE_DIR}/ttnn/ttnn.egg-info"
 )
+
+# mlir interface library settings
+set(MLIR_INTERFACE_LIB ttnn_mlir_interface)
+
+set(MLIR_INTERFACE_SOURCES
+    ${CMAKE_CURRENT_SOURCE_DIR}/cpp/ttnn/mlir_interface_api.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/cpp/ttnn/types_wrapper.cpp
+
+    ${CMAKE_CURRENT_SOURCE_DIR}/cpp/ttnn/common/op_constraints.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/cpp/ttnn/operations/eltwise/binary/binary_constraints.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/cpp/ttnn/operations/eltwise/unary/unary_constraints.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/cpp/ttnn/operations/matmul/matmul_constraints.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/cpp/ttnn/operations/normalization/softmax/softmax_constraints.cpp
+
+    # ${CMAKE_CURRENT_SOURCE_DIR}/cpp/ttnn/tensor/types.cpp                     # Shape
+    # ${CMAKE_CURRENT_SOURCE_DIR}/../tt_metal/impl/buffers/buffer.cpp         # Buffer
+    # ${CMAKE_CURRENT_SOURCE_DIR}/../tt_metal/tt_metal.cpp                   # tt::tt_metal::detail::AllocateBuffer
+    # ${CMAKE_CURRENT_SOURCE_DIR}/../tt_metal/impl/allocator/allocator.cpp   # tt::tt_metal::Device::logical_core_from_bank_id
+    # ${CMAKE_CURRENT_SOURCE_DIR}/../tt_metal/impl/device/device.cpp              # tt::tt_metal::Device::dram_channel_from_bank_id
+    # ${CMAKE_CURRENT_SOURCE_DIR}/../tt_metal/llrt/tt_cluster.cpp      # tt::Cluster::instance()
+    # ${CMAKE_CURRENT_SOURCE_DIR}/../tt_metal/llrt/llrt.cpp               # tt::llrt::write_hex_vec_to_core
+    # ${CMAKE_CURRENT_SOURCE_DIR}/../tt_metal/impl/device/device_pool.cpp # tt::DevicePool::_ins
+    # ${CMAKE_CURRENT_SOURCE_DIR}/../tt_metal/impl/program/program.cpp # tt::tt_metal::Program::compile
+    # this list is too long, linking ttnn.so for now
```

Comment:
I originally hoped we could get away from this, but leaving this problem for later.